moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator.cpp

```diff
@@ -352,10 +356,26 @@ bool TrajectoryGenerator::generate(const planning_scene::PlanningSceneConstPtr&
     return false;
   }
 
-  moveit::core::RobotState start_state(scene->getCurrentState());
-  moveit::core::robotStateMsgToRobotState(req.start_state, start_state, true);
-  setSuccessResponse(start_state, req.group_name, joint_trajectory, planning_begin, res);
+  setSuccessResponse(plan_info.start_scene->getCurrentState(), req.group_name, joint_trajectory, planning_begin, res);
   return true;
 }
 
+TrajectoryGenerator::MotionPlanInfo::MotionPlanInfo(const planning_scene::PlanningSceneConstPtr& scene,
+                                                    const planning_interface::MotionPlanRequest& req)
```

Comment:
```suggestion
                                                    const planning_interface::MotionPlanRequest& req) : start_scene(scene)
```

Reply:
> With this change I'm still getting z=0, rather than its value in the motion plan request.

That's the wrong start_scene. I will push a commit in a minute - if tests succeed locally.

> Are there any existing tests which use more than one joint model group?

Not sure. You could use the PR2 robot (having two arms), which is part of moveit_resources. Alternatively, use the Panda, which has a separate group for the hand and its finger joint.

Reply:
Am I right in thinking your intent was to generate `start_scene` from `start_state` and `start_joint_positions`, and then pass `start_scene` rather than `scene` to `extractMotionPlanInfo`?

Reply:
Exactly. I pushed a fixup for the missing initialization of `start_scene`. But there are still failing tests locally...